### DISCUSSION TITLE
Airflow Variable & Connection Usage by DAG

### DISCUSSION THREAD

**Discussion by JGoldman110** [apache/airflow#23427] [Q&A]
Can anyone recommend the best way to determine for a given airflow variable or connection, which DAGs are using it?

My Idea:

During task execution when an airflow variable or connection is requested, will get dag_id and task_id from context and make an entry into audit log table with event type as connection or variable. Saving the connection id or variable key in extras filed.

**Comment by potiuk** [comment]
This will be possible in the future if we move to DB-less workers and it will be planned by then. Se AIP-1  and underlying AIPs. https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-1%3A+Improve+Airflow+Security - before that it makes little sense as any of the access might be forged, so audit gives you nothing.
